### PR TITLE
Macros

### PR DESCRIPTION
**Pull request by PatrickDattilio**
Working macros 1-15. Looks like this also has the exit coordinate fixes, and the addition of the map  offset.

### CODE REVIEW

**Comment by PatrickDattilio** [comment]
#2 !

File: client/clientui.py [hunk 1]
```diff
@@ -270,46 +272,77 @@ def update_compass(self, compass_update):
...
+        tk.Button(macros, text='VIII', command=lambda: self.send_command("fe8")).grid(row=1, column=2, sticky='WENS')
+        tk.Button(macros, text='IX', command=lambda: self.send_command("fe9")).grid(row=1, column=3, sticky='WENS')
+        tk.Button(macros, text='X', command=lambda: self.send_command("fe10")).grid(row=1, column=4, sticky='WENS')
+
+        tk.Button(macros, text='XI', command=lambda: self.send_command("fe11")).grid(row=2, column=0, sticky='WENS')
+        tk.Button(macros, text='XII', command=lambda: self.send_command("fe12")).grid(row=2, column=1, sticky='WENS')
+        tk.Button(macros, text='XIII', command=lambda: self.send_command("fe13")).grid(row=2, column=2, sticky='WENS')
+        tk.Button(macros, text='XIV', command=lambda: self.send_command("fe14")).grid(row=2, column=3, sticky='WENS')
+        tk.Button(macros, text='XV', command=lambda: self.send_command("fe15")).grid(row=2, column=4, sticky='WENS')
+
```
Comment (marshall-davis) [inline client/clientui.py:337]: All the clients I see have 15 macro buttons and the commands `fe1` through `fe15` work. Going above that to `fe16` through `fe20` failed in my testing.

```
"fe20" matches no commands.
```

Comment (PatrickDattilio) [inline client/clientui.py:337]: Whoops. Got a little overzealous. Definitely should only be 1-15.

**Comment by PatrickDattilio** [comment]
All fixed now.

File: client/clientui.py (hunk 1 above)
Comment (marshall-davis) [inline client/clientui.py:337]: Haha, over "zealous". I see what you did there.

**Pull request merged**